### ISSUE TITLE
sqlfmt incorrectly format the field that contains $ character

### ISSUE THREAD

**Issue by edwintrinh** [tconbeer/sqlfmt#868]
**Describe the bug**
For external table in Snowflake, we can use some metadata fields like this:
```sql
select metadata$filename,
   metadata$file_last_modified
FROM external_table
```
but sqlfmt will format this as:

```sql
select metadata $filename,
   metadata $file_last_modified
from external_table
```
(adding extra space before $ character) that breaks the query

**To Reproduce**
create sql file with $ character in column name then format

**Expected behavior**
the column name should not be space-added.

**Actual behavior**
The space added before $ character break query

**Additional context**
sqlfmt, version 0.26.0

What is the output of `pip list` (or `pipx list` if you installed using pipx)?
installed via uv.

> Package                      Version     Editable project location
---------------------------- ----------- ----------------------------
adlfs                        2024.12.0
agate                        1.9.1
aiohappyeyeballs             2.6.1
aiohttp                      3.12.15
aiosignal                    1.4.0
alembic                      1.16.4
annotated-types              0.7.0
antlr4-python3-runtime       4.13.2
anyio                        4.12.0
asn1crypto                   1.5.1
astroid                      3.3.11
asttokens                    3.0.0
asyncssh                     2.21.1
attrs                        25.3.0
azure-core                   1.35.0
azure-datalake-store         0.0.53
azure-identity               1.23.1
azure-storage-blob           12.26.0
babel                        2.17.0
backoff                      2.2.1
base58                       2.1.1
bcrypt                       5.0.0
beautifulsoup4               4.13.5
black                        25.1.0
boruta                       0.4.3
boto3                        1.39.11
botocore                     1.39.11
bottle                       0.13.4
cachetools                   5.5.2
certifi                      2025.1.31
cffi                         1.17.1
charset-normalizer           3.4.2
click                        8.1.8
click-option-group           0.5.8
colorama                     0.4.6
coloredlogs                  14.0
colorlog                     6.12.0
contourpy                    1.3.3
cramjam                      2.11.0
cryptography                 45.0.5
cycler                       0.12.1
daff                         1.4.2
dagster                      1.12.1
dagster-dbt                  0.28.1
dagster-graphql              1.12.1
dagster-k8s                  0.28.1
dagster-pipes                1.12.1
dagster-postgres             0.28.1
dagster-sftp                 0.0.2
dagster-shared               1.12.1
dagster-slack                0.28.1
dagster-ssh                  0.28.1
dagster-webserver            1.12.1
dateparser                   1.4.2
dbt-adapters                 1.22.10
dbt-common                   1.37.5
dbt-core                     1.10.20
dbt-extractor                0.6.0
dbt-protos                   1.0.474
dbt-semantic-interfaces      0.9.0
dbt-snowflake                1.10.0
dbt-spark                    1.9.3
debugpy                      1.8.15
decorator                    5.2.1
deepdiff                     7.0.1
deprecated                   1.2.18
dill                         0.4.0
docstring-parser             0.17.0
dotty-dict                   1.3.1
duckdb                       1.3.2
durationpy                   0.10
et-xmlfile                   2.0.0
executing                    2.2.0
filelock                     3.18.0
fonttools                    4.62.0
frozenlist                   1.7.0
fsspec                       2025.7.0
future                       1.0.0
gitdb                        4.0.12
gitpython                    3.1.45
google-auth                  2.40.3
gql                          3.5.3
graphene                     3.4.3
graphql-core                 3.2.6
graphql-relay                3.2.0
greenlet                     3.2.3
grpcio                       1.80.0
grpcio-health-checking       1.80.0
grpcio-tools                 1.80.0
h11                          0.16.0
httptools                    0.6.4
humanfriendly                10.0
idna                         3.10
importlib-metadata           8.7.0
importlib-resources          6.5.2
iniconfig                    2.1.0
invoke                       2.2.1
ipdb                         0.13.13
ipython                      9.4.0
ipython-pygments-lexers      1.1.1
isodate                      0.6.1
isort                        6.0.1
jaraco-classes               3.4.0
jaraco-context               6.0.1
jaraco-functools             4.2.1
jedi                         0.19.2
jeepney                      0.9.0
jinja2                       3.1.6
jmespath                     1.0.1
joblib                       1.5.3
jsonschema                   4.25.0
jsonschema-specifications    2025.4.1
keyring                      25.6.0
kiwisolver                   1.5.0
kubernetes                   32.0.1
leather                      0.4.0
lxml                         6.0.1
mako                         1.3.10
markdown-it-py               3.0.0
markupsafe                   3.0.2
mashumaro                    3.14
matplotlib                   3.10.8
matplotlib-inline            0.1.7
mccabe                       0.7.0
mdurl                        0.1.2
more-itertools               10.7.0
msal                         1.33.0
msal-extensions              1.3.1
msgpack                      1.1.1
multidict                    6.6.3
mypy-extensions              1.1.0
narwhals                     2.24.0
networkx                     3.5
numpy                        2.3.2
nvidia-nccl-cu13             2.30.7
o365                         2.1.5
oauthlib                     3.3.1
office365-rest-python-client 2.6.2
openpyxl                     3.1.5
optuna                       4.9.0
optuna-dashboard             0.19.0
ordered-set                  4.1.0
orjson                       3.11.0
outcome                      1.3.0.post0
packaging                    25.0
pandas                       2.3.1
pandera                      0.26.0
paramiko                     4.0.0
parsedatetime                2.6
parso                        0.8.5
pathspec                     0.12.1
patsy                        1.0.2
pexpect                      4.9.0
pillow                       12.1.1
platformdirs                 4.3.8
pluggy                       1.6.0
prometheus-remote-writer     1.1.3
prompt-toolkit               3.0.51
propcache                    0.3.2
protobuf                     6.33.6
psycopg2-binary              2.9.10
ptyprocess                   0.7.0
pure-eval                    0.2.3
pure-sasl                    0.6.2
pyarrow                      18.1.0
pyasn1                       0.6.1
pyasn1-modules               0.4.2
pycparser                    2.22
pydantic                     2.11.7
pydantic-core                2.33.2
pydantic-settings            2.10.1
pygments                     2.19.2
pyhive                       0.7.0
pyjwt                        2.10.1
pylint                       3.3.8
pylint-plugin-utils          0.9.0
pylint-pydantic              0.3.5
pynacl                       1.6.0
pyopenssl                    25.1.0
pyparsing                    3.3.2
pysocks                      1.7.1
pytest                       8.4.2
python-dateutil              2.9.0.post0
python-dotenv                1.1.1
python-gitlab                6.4.0
python-semantic-release      10.4.1
python-slugify               8.0.4
python-snappy                0.7.3
pytimeparse                  1.1.8
pytz                         2025.2
pyyaml                       6.0.2
referencing                  0.36.2
regex                        2026.7.19
requests                     2.33.1
requests-oauthlib            2.0.0
requests-toolbelt            1.0.0
rich                         14.0.0
rpds-py                      0.26.0
rsa                          4.9.1
ruff                         0.12.4
s3transfer                   0.13.1
scikit-learn                 1.9.0
scipy                        1.18.0
secretstorage                3.3.3
selenium                     4.32.0
setuptools                   80.9.0
shandy-sqlfmt                0.26.0
shellingham                  1.5.4
six                          1.17.0
slack-sdk                    3.37.0
smmap                        5.0.2
sniffio                      1.3.1
snowflake-connector-python   3.16.0
snowplow-tracker             1.1.0
sortedcontainers             2.4.0
soupsieve                    2.7
sqlalchemy                   2.0.41
sqlglot                      27.2.0
sqlglotrs                    0.6.1
sqlparams                    6.2.0
sqlparse                     0.5.3
sshfs                        2025.11.0
sshtunnel                    0.4.0
stack-data                   0.6.3
starlette                    0.47.2
statsmodels                  0.14.6
structlog                    25.4.0
tabulate                     0.9.0
text-unidecode               1.3
threadpoolctl                3.6.0
thrift                       0.16.0
thrift-sasl                  0.4.3
tomli                        2.2.1
tomlkit                      0.13.3
toposort                     1.10
tqdm                         4.67.1
traitlets                    5.14.3
trio                         0.31.0
trio-websocket               0.12.2
typeguard                    4.4.4
typer                        0.16.0
typing-extensions            4.14.1
typing-inspect               0.9.0
typing-inspection            0.4.1
tzdata                       2025.2
tzlocal                      5.3.1
universal-pathlib            0.2.6
urllib3                      2.5.0
uvicorn                      0.35.0
uvloop                       0.21.0
watchdog                     5.0.3
watchfiles                   1.1.0
wcwidth                      0.2.13
webdriver-manager            4.0.2
websocket-client             1.8.0
websockets                   15.0.1
wrapt                        1.17.3
wsproto                      1.2.0
xgboost                      3.4.0
xlrd                         2.0.2
yarl                         1.20.1
zipp                         3.23.0

**Comment by tconbeer** [comment]
This has been fixed. Please upgrade to receive it.
